### ISSUE TITLE
Backport: Adding private registry in GUI prone to causing cluster creation issues

### ISSUE THREAD

**Issue by westlywright** [rancher/rancher#19345]
Backport: rancher/rancher#19325

**Comment by jira-sync-svc** [comment]
➤ David Nuzik commented:

Just like ER-92 is blocked by ER-336

**Comment by jira-sync-svc** [comment]
➤ David Nuzik commented:

Closed blocking issue #19734 (Github) it is actually a non-issue and is not blocking.

**Comment by alena1108** [comment]
can be tested in rc9

**Comment by davidnuzik** [comment]
Version: v2.2.4-rc1

The UI for Private Registries for cluster config has been fixed (see the PR). We removed the option to have more than one entry because only one should be used. Also, this one is always default. It's for system-images and the help text under Private Registries accordion explains this. (%cruPrivateRegistry.title.label%)

![image](https://user-images.githubusercontent.com/45179589/57575091-25d0c280-73f9-11e9-95ec-f9affcd8e222.png)

Notice the subtext for "Private Registries" this explains what the registry is for.

Once Enabled you can config the registry url, user, and password just like before. I validated the fields and ensured HTTP and API are correct and enforce their respective types. (There is one issue though mentioned below which is NOT related to this UI fix).

![image](https://user-images.githubusercontent.com/45179589/57575101-5b75ab80-73f9-11e9-8687-5010c3bc7ef6.png)


It looks like isDefault has changed in the API schema. isDefault for privateRegistries is now false instead of true. @alena1108 may I ask to you happen to know why this may have changed?

http:
```
privateRegistries: [,…]
0: {isDefault: false, type: "/v3/schemas/privateRegistry", url: "https://testurl.com", user: "testuser"}
isDefault: false
type: "/v3/schemas/privateRegistry"
url: "https://testurl.com"
user: "testuser"
```

schema:
```
isDefault": {
"create": true,
"default": false,
"type": "boolean",
"update": true
},
```

Since this is working correctly from a  UI standpoint I will be closing the issue. If this API schema change was not expected we will open another issue if necessary.

**Comment by davidnuzik** [comment]
I am re-opening this issue. There was confusion around what the default should be. The UI should be setting isDefault to true but does not appear to be doing this.

I need to test again

**Comment by alena1108** [comment]
Can be tested with `v2.2.4-rc6`

**Comment by davidnuzik** [comment]
Version: v2.2.4-rc6

isDefault is now overridden by the UI. It is now set to true by the UI.
```
privateRegistries: [{isDefault: true, type: "/v3/schemas/privateRegistry", url: "https://test.com", user: "testuser"}]
0: {isDefault: true, type: "/v3/schemas/privateRegistry", url: "https://test.com", user: "testuser"}
isDefault: true
type: "/v3/schemas/privateRegistry"
url: "https://test.com"
user: "testuser"
```

The issue is resolved. Refer to my comment above for my testing done earlier. All checks still pass I just validated that isDefault is now true. https://github.com/rancher/rancher/issues/19345#issuecomment-492338193
